### PR TITLE
ENH: Make Tables module translatable (SlicerLanguagePacks#12)

### PR DESCRIPTION
**Pull request by Papa96108**

### CODE REVIEW

File: Libs/MRML/Core/vtkMRMLTableNode.cxx [hunk 1]
```diff
@@ -981,13 +981,27 @@ int vtkMRMLTableNode::GetColumnValueTypeFromSchema(const std::string& columnName
 //----------------------------------------------------------------------------
 void vtkMRMLTableNode::SetColumnLongName(const std::string& columnName, const std::string& longName)
 {
-  this->SetColumnProperty(columnName, SCHEMA_COLUMN_LONG_NAME, longName);
+  vtkWarningMacro("This method is deprecated and will be unavailable soon. Please use SetColumnTitle instead");
```
Comment (jcfr) [inline Libs/MRML/Core/vtkMRMLTableNode.cxx:984]:
```suggestion
  vtkWarningMacro("vtkMRMLTableNode::SetColumnLongName is deprecated, use vtkMRMLSequenceBrowserNode::SetColumnTitle method instead");
```

**Review by jcfr — CHANGES_REQUESTED** [review]
Overall the changes look good.

Before integrating, I suggest to have two commits:
- `ENH: Improve TableNode API for column title`
- `ENH: Make Table module translatable`
` `ENH: Make SegmentStatistics measurement translatable`

### Questions

* Will the update changing `SCHEMA_COLUMN_LONG_NAME` to `SCHEMA_COLUMN_TITLE` impact loading of table saved using the older column name ?


### Notes

When referencing issues outside of the Slicer project make sure to use the format `Organization/Repository#IssueNumber`, failure to do so is referencing issue in the "wrong" project.

See more details at https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls

**Comment by Papa96108** [comment]
Hi @jcfr  thanks for your suggestion should I rewrite my commit message for commit message you suggest with git rebase?

**Comment by Papa96108** [comment]
@jcfr please run the workflow

**Comment by Papa96108** [comment]
The revert commit message always causes the check commit message workflow to fail can you help me find a solution i've already used git rebase to change the commit message without success. 

https://github.com/Slicer/Slicer/commit/0d06e8deeffaf2a4d21947f830dd9131537f4c77.
@lassoan 
@pieper 
@jcfr 
@mhdiop

**Comment by mhdiop** [comment]
Thank you @Papa96108  for working on this issue.

We can have a Meet tonight to discuss about the difficulties you are facing.

Regards.

**Comment by Papa96108** [comment]
> Thank you @Papa96108 for working on this issue.
> 
> We can have a Meet tonight to discuss about the difficulties you are facing.
> 
> Regards.

Thanks you @mhdiop see you tonight

**Comment by lassoan** [comment]
I've reviewed and fixed up the code, it all looks good now. Modules that provide translatable column name can enable `UseColumnTitleAsColumnHeader` to display that in the header (instead of displaying the column name).

> * Will the update changing `SCHEMA_COLUMN_LONG_NAME` to `SCHEMA_COLUMN_TITLE` impact loading of table saved using the older column name ?

I've added `ReadLongNameAsTitle` flag to allow using `longName` as `title`, but since often longName is too long for title, it is disabled by default. The only known use of longName is in Segment Statistics result tables, and there longName is too long.

**Comment by lassoan** [comment]
@jcfr You have requested changes earlier, please have a look and approve if it looks good to you.

**Comment by jcfr** [comment]
@lassoan Would it be sensible to integrate this one into `5.6.1` ?

**Comment by lassoan** [comment]
Yes, I think so.

**Pull request merged**